### PR TITLE
Add Hide Notification hyperlink to Notification Panel

### LINKED ISSUE #3708: Allow hiding "Flutter commands"
## Steps to Reproduce

Please allow to turn the "Flutter commands of" for certain messages:

image

I am working in IntelliJ IDEA and I have this message constantly "Open for editing in Android Studio" even though I got gradle to work and Android Studio is useless and I want to hide that message.

## Version info

```
[√] Android Studio (version 3.4)
    • Flutter plugin version 37.1.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[√] IntelliJ IDEA Ultimate Edition (version 2019.2)
    • IntelliJ at F:\PROGRAMY\jetBRAINS\IDEA\IntelliJ IDEA 2018.3.2
    • Flutter plugin version 37.1.4
    • Dart plugin version 192.5728.98

```

### PR DESCRIPTION
**Pull request by frextrite**
Closes #3708 

The changes should theoretically work though I was unable to test the changes due to some errors in setting-up my development environment. Working on fixing that.

### CODE REVIEW

**Comment by frextrite** [comment]
@pq @devoncarew need review

**Comment by devoncarew** [comment]
Thanks for the PR! I'll pass to @pq for review (he's OOO currently, so there may be a slight delay).

**Comment by frextrite** [comment]
Sure, thank you.

**Comment by frextrite** [comment]
I thought the same, hence the method I currently use, stores the information during the runtime. So even if `Hide notification` is pressed, the notification would be shown again the next time IDE is relaunched. For persisting the information maybe we could use `PropertiesComponent`?

**Comment by frextrite** [comment]
Last I tried I still had the issues. I'll try once again today and revert back.

**Comment by frextrite** [comment]
I got the plugin to work, how do I trigger the notification panel manually for checking purposes?

**Comment by frextrite** [comment]
I'm running into something strange now. The test IDE that opens doesn't show `Flutter` plugin under the installed plugins section. It shows `Dart` however.

Btw, I just realized you were checking the `NotificationPanel` created by `FlutterSampleNotificationProvider`. The change I did (and thought was intended) shows `Hide Notification` only in panels created by `NativeEditorNotificationProvider` (related to opening of the project in Android Studio if it is present)

**Comment by frextrite** [comment]
I was trying to set up the IDE for development, still no luck so far. I created a new `runIde` configuration as
```
runIde {
  ideaDirectory 'C:\\Users\\username\\AppData\\Local\\JetBrains\\Toolbox\\apps\\IDEA-U\\ch-0\\192.6262.58'
}
```
When manually invoking this configuration from `Gradle` menu, I'm getting the following exceptions,

```
com.intellij.diagnostic.PluginException:  [Plugin: io.flutter]
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.instantiate(ActionManagerImpl.java:163)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.convertStub(ActionManagerImpl.java:139)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.getActionImpl(ActionManagerImpl.java:501)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.getAction(ActionManagerImpl.java:489)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.preloadActions(ActionManagerImpl.java:1327)
	at com.intellij.openapi.actionSystem.impl.ActionPreloader.preload(ActionPreloader.java:18)
	at com.intellij.openapi.application.Preloader.lambda$null$1(Preloader.java:74)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:169)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:156)
	at com.intellij.openapi.application.Preloader.lambda$componentsInitialized$2(Preloader.java:71)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:207)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$100(BoundedTaskExecutor.java:29)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.lambda$run$0(BoundedTaskExecutor.java:185)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:208)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:181)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ExceptionInInitializerError
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.instantiate(ActionManagerImpl.java:151)
	... 20 more
Caused by: java.util.MissingResourceException: Can't find bundle for base name io.flutter.FlutterBundle, locale en_IN
	at java.util.ResourceBundle.throwMissingResourceException(ResourceBundle.java:1581)
	at java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1396)
	at java.util.ResourceBundle.getBundle(ResourceBundle.java:782)
	at io.flutter.FlutterBundle.getBundle(FlutterBundle.java:32)
	at io.flutter.FlutterBundle.message(FlutterBundle.java:23)
	at io.flutter.actions.RunReleaseFlutterApp.<clinit>(RunReleaseFlutterApp.java:14)
	... 23 more
Caused by: java.lang.ClassCastException: io.flutter.FlutterBundle cannot be cast to ResourceBundle
	at java.util.ResourceBundle$Control.newBundle(ResourceBundle.java:2664)
	at java.util.ResourceBundle.loadBundle(ResourceBundle.java:1518)
	at java.util.ResourceBundle.findBundle(ResourceBundle.java:1482)
	at java.util.ResourceBundle.findBundle(ResourceBundle.java:1436)
	at java.util.ResourceBundle.findBundle(ResourceBundle.java:1436)
	at java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1370)
	... 27 more
```

and

```
com.intellij.openapi.extensions.impl.PicoPluginExtensionInitializationException: java.lang.ClassNotFoundException: com.jetbrains.lang.dart.ide.runner.DartLineBreakpointType PluginClassLoader[Dart, 192.6262.58] com.intellij.ide.plugins.cl.PluginClassLoader@75062575 [Plugin Dart]
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.createInstance(ExtensionComponentAdapter.java:50)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.createInstance(XmlExtensionAdapter.java:54)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapter(ExtensionPointImpl.java:439)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapters(ExtensionPointImpl.java:410)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.getExtensionList(ExtensionPointImpl.java:226)
	at com.intellij.openapi.extensions.ExtensionPointName.getExtensionList(ExtensionPointName.java:39)
	at com.intellij.xdebugger.impl.breakpoints.XBreakpointUtil.breakpointTypes(XBreakpointUtil.java:60)
	at com.intellij.xdebugger.impl.breakpoints.XBreakpointManagerImpl.<init>(XBreakpointManagerImpl.java:61)
	at com.intellij.xdebugger.impl.XDebuggerManagerImpl.<init>(XDebuggerManagerImpl.java:99)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.picocontainer.defaults.InstantiatingComponentAdapter.newInstance(InstantiatingComponentAdapter.java:193)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.doGetComponentInstance(CachingConstructorInjectionComponentAdapter.java:88)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.instantiateGuarded(CachingConstructorInjectionComponentAdapter.java:66)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getComponentInstance(CachingConstructorInjectionComponentAdapter.java:48)
	at com.intellij.openapi.components.impl.ComponentManagerImpl$ComponentConfigComponentAdapter.getComponentInstance(ComponentManagerImpl.java:481)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.createComponents(ComponentManagerImpl.java:126)
	at com.intellij.openapi.project.impl.ProjectImpl.init(ProjectImpl.java:262)
	at com.intellij.openapi.project.impl.ProjectManagerImpl.initProject(ProjectManagerImpl.java:285)
	at com.intellij.openapi.project.impl.ProjectManagerImpl.access$300(ProjectManagerImpl.java:63)
	at com.intellij.openapi.project.impl.ProjectManagerImpl$2.run(ProjectManagerImpl.java:545)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:894)
	at com.intellij.openapi.progress.impl.CoreProgressManager$5.run(CoreProgressManager.java:447)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:169)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:156)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$null$9(ApplicationImpl.java:552)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:294)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.intellij.openapi.extensions.ExtensionInstantiationException: java.lang.ClassNotFoundException: com.jetbrains.lang.dart.ide.runner.DartLineBreakpointType PluginClassLoader[Dart, 192.6262.58] com.intellij.ide.plugins.cl.PluginClassLoader@75062575 [Plugin Dart]
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getImplementationClass(ExtensionComponentAdapter.java:94)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.createInstance(ExtensionComponentAdapter.java:38)
	... 36 more
Caused by: java.lang.ClassNotFoundException: com.jetbrains.lang.dart.ide.runner.DartLineBreakpointType PluginClassLoader[Dart, 192.6262.58] com.intellij.ide.plugins.cl.PluginClassLoader@75062575
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:75)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getImplementationClass(ExtensionComponentAdapter.java:91)
	... 37 more
Caused by:

com.intellij.openapi.extensions.ExtensionInstantiationException: java.lang.ClassNotFoundException: com.jetbrains.lang.dart.ide.runner.DartLineBreakpointType PluginClassLoader[Dart, 192.6262.58] com.intellij.ide.plugins.cl.PluginClassLoader@75062575 [Plugin Dart]
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getImplementationClass(ExtensionComponentAdapter.java:94)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.createInstance(ExtensionComponentAdapter.java:38)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.createInstance(XmlExtensionAdapter.java:54)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapter(ExtensionPointImpl.java:439)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapters(ExtensionPointImpl.java:410)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.getExtensionList(ExtensionPointImpl.java:226)
	at com.intellij.openapi.extensions.ExtensionPointName.getExtensionList(ExtensionPointName.java:39)
	at com.intellij.xdebugger.impl.breakpoints.XBreakpointUtil.breakpointTypes(XBreakpointUtil.java:60)
	at com.intellij.xdebugger.impl.breakpoints.XBreakpointManagerImpl.<init>(XBreakpointManagerImpl.java:61)
	at com.intellij.xdebugger.impl.XDebuggerManagerImpl.<init>(XDebuggerManagerImpl.java:99)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.picocontainer.defaults.InstantiatingComponentAdapter.newInstance(InstantiatingComponentAdapter.java:193)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.doGetComponentInstance(CachingConstructorInjectionComponentAdapter.java:88)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.instantiateGuarded(CachingConstructorInjectionComponentAdapter.java:66)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getComponentInstance(CachingConstructorInjectionComponentAdapter.java:48)
	at com.intellij.openapi.components.impl.ComponentManagerImpl$ComponentConfigComponentAdapter.getComponentInstance(ComponentManagerImpl.java:481)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.createComponents(ComponentManagerImpl.java:126)
	at com.intellij.openapi.project.impl.ProjectImpl.init(ProjectImpl.java:262)
	at com.intellij.openapi.project.impl.ProjectManagerImpl.initProject(ProjectManagerImpl.java:285)
	at com.intellij.openapi.project.impl.ProjectManagerImpl.access$300(ProjectManagerImpl.java:63)
	at com.intellij.openapi.project.impl.ProjectManagerImpl$2.run(ProjectManagerImpl.java:545)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:894)
	at com.intellij.openapi.progress.impl.CoreProgressManager$5.run(CoreProgressManager.java:447)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:169)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:156)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$null$9(ApplicationImpl.java:552)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:294)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassNotFoundException: com.jetbrains.lang.dart.ide.runner.DartLineBreakpointType PluginClassLoader[Dart, 192.6262.58] com.intellij.ide.plugins.cl.PluginClassLoader@75062575
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:75)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getImplementationClass(ExtensionComponentAdapter.java:91)
	... 37 more
```

**Comment by pq** [comment]
@frextrite: sorry for the misunderstanding! I just tested your change locally and it looks good.

image

The label is a bit verbose but I can't think of anything better off the top of my head (maybe @InMatrix has thoughts?) @alexander-doroshko: are there examples elsewhere we could be consistent with?

One possible tweak would be a detail in the hover text that explains this will dismiss the notification until restart?  (Or similar.)

image

Sorry about your continued woes testing the plugin.  It's been a while since I set my environment up but this part doesn't seem right:

> When manually invoking this configuration from Gradle menu, I'm getting the following exceptions,

(I'm sure I've never used gradle.) Maybe @stevemessick has quick ideas?

**Comment by stevemessick** [comment]
@frextrite You should need to do anything with Gradle to get IntelliJ set up to work on the Flutter plugin. There *is* a sub-module named flutter-gui-tests that requires Gradle. It is used to run integration tests. The only reason there are Gradle build files at the top level is to support that module. If you follow the instructions in CONTRIBUTING.md you should be up and running in a few minutes.

One reason you might not see the Flutter plugin when you launch IntelliJ in the debugger is if the plugin.xml specifies an upper bound for versions that is too low. I think the current plugin.xml has 192.* as its upper bound. That needs to be 193.* in IntelliJ 2019.2.1.

**Comment by alexander-doroshko** [comment]
> @alexander-doroshko: are there examples elsewhere we could be consistent with?

I'm not aware of any 'standard', I think 'Hide notification' is fine.

**Comment by InMatrix** [comment]
> One possible tweak would be a detail in the hover text that explains this will dismiss the notification until restart? (Or similar.)

This sounds reasonable. The label can probably be shortened to just "Hide" with a tooltip explaining this type of notification will be hidden until restart.

**Comment by pq** [comment]
> The label can probably be shortened to just "Hide" with a tooltip explaining this type of notification will be hidden until restart.

SGTM. @frextrite: if you want to make that change let me know.  Otherwise, I can land and follow-up.

Thanks!

**Comment by frextrite** [comment]
> > The label can probably be shortened to just "Hide" with a tooltip explaining this type of notification will be hidden until restart.
> 
> SGTM. @frextrite: if you want to make that change let me know. Otherwise, I can land and follow-up.
> 
> Thanks!

@pq I'll make the change. The notification label as `Hide` and the tooltip text, `Hide notification until IDE restart`. Does this sound good?

**Comment by frextrite** [comment]
@stevemessick Thanks for the information. I didn't know Gradle was just being used for running GUI tests. I assumed it was used for running the plugin since I followed that workflow for my project. I'll give it a shot and get back to you.

**Comment by pq** [comment]
That change sounds great.  Thanks @frextrite !

File: src/io/flutter/FlutterBundle.properties [hunk 1]
```diff
@@ -199,8 +199,8 @@ flutter.androidstudio.open.module.text=Open Android module in Android Studio
 flutter.androidstudio.open.module.description=Open Android Studio in a new window to edit the top-level Android project
 flutter.androidstudio.open.file.text=Open for Editing in Android Studio
 flutter.androidstudio.open.file.description=Open Android Studio in a new window to edit this file
-flutter.androidstudio.open.hide.notification.text=Hide notification
-flutter.androidstudio.open.hide.notification.description=Hide notification
+flutter.androidstudio.open.hide.notification.text=Hide
+flutter.androidstudio.open.hide.notification.description=Hide notification until IDE restart
```
Comment (pq) [inline src/io/flutter/FlutterBundle.properties]: 👍

**Comment by frextrite** [comment]
> Relatedly, @frextrite: I'll want to update to AUTHORS as well to give you attribution... What's a good email?

Awesome, thank you! You can use my primary `frextrite[at]gmail.com`. Looking forward to contribute more!

**Comment by stevemessick** [comment]
Thanks @frextrite it is great to see more people getting involved :)

**Comment by pq** [comment]
@frextrite: have you submitted our   [Contributor License Agreement](https://cla.developers.google.com/clas)?

**Comment by frextrite** [comment]
@pq Yes, I have.

@stevemessick Thank you for your help. I have successfully set up the dev environment, flutter plugin is working without any issues.

**Pull request merged**